packages/drivers/driver-web-cache/src/FluidCache.ts

```diff
@@ -117,10 +120,35 @@ export class FluidCache implements IPersistedCache {
 		});
 	}
 
+	private async openDb() {
+		if (this.db === undefined) {
+			this.db = await getFluidCacheIndexedDbInstance(this.logger);
+			// Need to close the db on version change if opened.
+			this.db.addEventListener("versionchange", (ev) => {
+				this.db?.close();
+				this.db = undefined;
+			});
+			this.db.addEventListener("close", (ev) => {
+				clearTimeout(this.dbCloseTimer);
+				this.dbCloseTimer = undefined;
+				this.db = undefined;
+			});
+			// Schedule db close after 30s.
+			if (this.dbCloseTimer === undefined) {
+				this.dbCloseTimer = setTimeout(() => {
+					this.db?.close();
+					this.db = undefined;
+					this.dbCloseTimer = undefined;
+				}, 30000);
+			}
+		}
+		return this.db;
+	}
+
 	public async removeEntries(file: IFileEntry): Promise<void> {
 		let db: IDBPDatabase<FluidCacheDBSchema> | undefined;
 		try {
-			db = await getFluidCacheIndexedDbInstance(this.logger);
+			db = await this.openDb();
 
```

Comment:
out of curiosity, can the db close immediately after we getFluidCacheIndexedDbInstance ? Ex. event close ? If so, is it possible db can return undefined ?

Reply:
What do you mean by db return undefined? Do you mean that openDB, function return undefined? It should not close automatically and if it does, then we have try/catch in functions using the db and we will just catch the expection and return undefined.